### DISCUSSION TITLE
Accessing methods in the shadow dom element from outside - LIT 2.2

### DISCUSSION THREAD

**Discussion by OvidijusParsiunas** [lit/lit#3585] [Q&A] [answered]
Is there a way to call a method inside the custom element from outside?

I have searched far and wide for documentation regarding this but cannot find anything related and from what I could scrap on the web, the following should work, but it returns the following error:
```Uncaught TypeError: element.customMethod is not a function```

TS:
```
import {html, css, LitElement} from 'lit';
import {customElement, property} from 'lit/decorators.js';

@customElement('simple-greeting')
export class SimpleGreeting extends LitElement {  
  customMethod() {
    console.log('custom method was called');
  }

  render() {
    return html`<p>Check the Console to see the logged method call</p>`;
  }
}
```

 HTML:
 ```
<!DOCTYPE html>
<head>
  <script type="module" src="./simple-greeting.js"></script>
</head>
<body>
 <simple-greeting id='element'></simple-greeting>
  <script>
    var element = document.getElementById('element');
    element.customMethod();
  </script>
</body>
```

[Link to the playground](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy50cyIsImNvbnRlbnQiOiJpbXBvcnQge2h0bWwsIGNzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7Y3VzdG9tRWxlbWVudCwgcHJvcGVydHl9IGZyb20gJ2xpdC9kZWNvcmF0b3JzLmpzJztcblxuQGN1c3RvbUVsZW1lbnQoJ3NpbXBsZS1ncmVldGluZycpXG5leHBvcnQgY2xhc3MgU2ltcGxlR3JlZXRpbmcgZXh0ZW5kcyBMaXRFbGVtZW50IHsgIFxuICBjdXN0b21NZXRob2QoKSB7XG4gICAgY29uc29sZS5sb2coJ2N1c3RvbSBtZXRob2Qgd2FzIGNhbGxlZCcpO1xuICB9XG5cbiAgcmVuZGVyKCkge1xuICAgIHJldHVybiBodG1sYDxwPkNoZWNrIHRoZSBDb25zb2xlIHRvIHNlZSB0aGUgbG9nZ2VkIG1ldGhvZCBjYWxsPC9wPmA7XG4gIH1cbn1cbiJ9LHsibmFtZSI6ImluZGV4Lmh0bWwiLCJjb250ZW50IjoiPCFET0NUWVBFIGh0bWw-XG48aGVhZD5cbiAgPHNjcmlwdCB0eXBlPVwibW9kdWxlXCIgc3JjPVwiLi9zaW1wbGUtZ3JlZXRpbmcuanNcIj48L3NjcmlwdD5cbjwvaGVhZD5cbjxib2R5PlxuIDxzaW1wbGUtZ3JlZXRpbmcgaWQ9J2VsZW1lbnQnPjwvc2ltcGxlLWdyZWV0aW5nPlxuICA8c2NyaXB0PlxuICAgIHZhciBlbGVtZW50ID0gZG9jdW1lbnQuZ2V0RWxlbWVudEJ5SWQoJ2VsZW1lbnQnKTtcbiAgICBlbGVtZW50LmN1c3RvbU1ldGhvZCgpO1xuICA8L3NjcmlwdD5cbjwvYm9keT5cbiJ9LHsibmFtZSI6InBhY2thZ2UuanNvbiIsImNvbnRlbnQiOiJ7XG4gIFwiZGVwZW5kZW5jaWVzXCI6IHtcbiAgICBcImxpdFwiOiBcIl4yLjAuMFwiLFxuICAgIFwiQGxpdC9yZWFjdGl2ZS1lbGVtZW50XCI6IFwiXjEuMC4wXCIsXG4gICAgXCJsaXQtZWxlbWVudFwiOiBcIl4zLjAuMFwiLFxuICAgIFwibGl0LWh0bWxcIjogXCJeMi4wLjBcIlxuICB9XG59IiwiaGlkZGVuIjp0cnVlfV0)

Am I doing something wrong or is calling methods from outside of the component not supported?

Thankyou!

**Comment by e111077** [comment, marked as answer]
This should work. The problem is the timing of your scripts. You have the element's definition in a `script[type=module]` in `<head>`. `type[module]` will defer the script until after the page has been parsed. meanwhile you call the element method in an inline script that does not have a type on it or any other timing modifiers. That means that it will be parsed and run as the page is streamed and thus runs BEFORE your element is defined.

**TL;DR**, add `type="module"` to the second script.

If you wish to learn more, [this article](https://levelup.gitconnected.com/html-script-element-attributes-async-vs-defer-vs-type-module-610b50a79dbd) seems to cover the topic pretty well.

**Reply by OvidijusParsiunas** [reply]
Wow, been working with JavaScript for so many years but had no idea this is how scripts are loaded, thankyou so much!
